commit 1d8e3dda4e37a41003a78b39818041a6b7c38452
Author: fonsinchen <ulfonk_mennhar@gmx.de>
Date:   Mon Mar 4 11:18:59 2013 +0100

    -Fix: don't try to unload if there isn't anything to unload

diff --git a/src/economy.cpp b/src/economy.cpp
index e32ffa3..8a4e535 100644
--- a/src/economy.cpp
+++ b/src/economy.cpp
@@ -1220,7 +1220,7 @@ void PrepareUnload(Vehicle *front_v)
 				v->cargo.Stage(
 						HasBit(Station::Get(front_v->last_station_visited)->goods[v->cargo_type].acceptance_pickup, GoodsEntry::GES_ACCEPTANCE),
 						front_v->last_station_visited, front_v->current_order.GetUnloadType());
-				SetBit(v->vehicle_flags, VF_CARGO_UNLOADING);
+				if (v->cargo.UnloadCount() > 0) SetBit(v->vehicle_flags, VF_CARGO_UNLOADING);
 			}
 		}
 	}
